cypress-io/cypress
# fix: vite v3 not working with node >=17

- Closes #23042 

### User facing changelog
CT projects using Vite v3 work with Node versions >=17, and a comma is removed from react + vite 3 apps in index.html

### Additional details
We now force the host to be `127.0.0.1` when spinning up the vite-dev-server. We did the same fix for webpack a while ago (https://github.com/cypress-io/cypress/pull/21430) and Vite v3 needs a similar fix.

I also fixed an issue that is only prevalent when using vite + react where a comma was being inserted into the served `index.html` due to how we were injecting scripts.

I added a vite setup for the `simple-ct` system test proejct and added it to the binary tests to verify behavior across node versions.

### Steps to test
This issue can be seen inside our own codebase, just switch to a node version >= 17 and try running any our our vite component tests. See the original issue for repro steps, you can now point the locally running cypress at a vite 3 project using a node version >= 17 and it will now work.

### How has the user experience changed?
Before: (Notice the comma on the second load when I switch back to Node 16)

https://user-images.githubusercontent.com/25158820/182245411-6edafed7-8aa3-4f7a-a1a8-4e8cc25dff2f.mov

After:

https://user-images.githubusercontent.com/25158820/182245539-9a2a029f-0c10-4663-a5e3-446b451e82f7.mov


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
